Provide documentation on configuration profiles/components/add ons. (#3620)
istio/istio.io: content/docs/setup/kubernetes/additional-setup/config-profile/index.md

---

> title: Configuration Profiles

Reviewer: Installation Configuration Profiles

Author: Updated

> description: Describe the provided configuration profiles.

Author: Done

weight: 35
keywords: [profiles]
---

> This page describes the provided helm configuration profiles. The profiles provide customization of the Istio control plane and of the sidecars for the Istio data plane. You can start with one of Istio’s built-in configuration profiles and then further customize the configuration for your specific needs. The following built-in configuration profiles are currently available:

Reviewer: ```suggestion
This page describes the built-in configuration profiles that can be used when
[installing Istio using helm](/docs/setup/kubernetes/install/kubernetes/).
The profiles provide customization of the Istio control plane and of the sidecars for the Istio data plane.
You can start with one of Istio’s built-in configuration profiles and then further customize the configuration for
your specific needs. The following built-in configuration profiles are currently available:
```

Author: Done

| `istio-citadel` | X | X |   | X | X |
| Gateways |   |   |   |   |  |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-egressgateway` |   | X |   |   | |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-ingressgateway` | X | X |   | X | X |
| `istio-galley` | X | X |   | X | X |
| `istio-pilot` | X | X | X |   | X |
| `istio-sidecar-injector` | X | X |   | X | X |
| `istio-telemetry` | X | X |   |   | X |
| `istio-nodeagent` |   |   |   |   | X |
| Add on |   |   |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`prometheus` | X | X |   |   | X |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-tracing` |   | X |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`grafana` |   | X |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`kiali` |   | X |   |   |   |
|   |   |   |   |   |   |
| The authentication version, adding `-auth` adds |   |   |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Control Plane Security |   | X |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mTLS |   | X |   |   | X |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;SDS |   |   |   |   | X |

> To further customize Istio and install addons, you can add one or more `--set <key>=<value>` options in the`helm template` or `helm install` command in the instructions, below. The [Installation Options](/docs/reference/config/installation-options/) lists the complete set of supported installation key and value pairs.

Reviewer: ```suggestion
To further customize Istio and install addons, you can add one or more `--set <key>=<value>` options in the`helm template` or `helm install` command that you use when installing Istio. The [Installation Options](/docs/reference/config/installation-options/) lists the complete set of supported installation key and value pairs.
```

Author: Done

    that you can more easily enable additional features if you wish to explore more advanced tasks. This profile comes in two flavors, either with or without authentication enabled.

1. **minimal**: the minimal set of components necessary to use Istio's [traffic management](/docs/tasks/traffic-management/) features.

1. **remote**: creates a service account with minimal access for use by Istio Pilot discovery used in [configuring a multicluster mesh](/docs/examples/multicluster/split-horizon-eds/).

1. **sds**:  used to enable [SDS (secret discovery service) for Istio](/docs/tasks/security/auth-sds). This profile comes only with authentication enabled.

The components marked as **X** are installed within each profile:

| Component | default | demo | minimal | remote | sds |
| --- | :---: | :---: | :---: | :---: | :---: |
|Profile filename | `values.yaml` | `values-istio-demo.yaml` | `values-istio-minimal.yaml` | `values-istio-remote.yaml` | `values-istio-sds-auth.yaml` |
| `istio-citadel` | X | X |   | X | X |
| Gateway |   |   |   |   |  |

> | &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-egressgateway` |   | X |   |   | |

Author: Remove galley and istio-gateway from remote profile.

Author: Done

> ---

Reviewer 2: Should the directory be config-profiles?

Author: Done

title: Installation Configuration Profiles
description: Describes the built-in Istio installation configuration profiles.
weight: 35

> keywords: [profiles]

Reviewer 2: Maybe a few more keywords to improve "see also". Look at other files for any keywords that . would make sense here.

Reviewer: install,helm

Author: Done

---
title: Installation Configuration Profiles
description: Describes the built-in Istio installation configuration profiles.
weight: 35
keywords: [profiles]
---

This page describes the built-in configuration profiles that can be used when

> [installing Istio using helm](/docs/setup/kubernetes/install/kubernetes/).

Reviewer: ```suggestion
[installing Istio using helm](/docs/setup/kubernetes/install/helm/).
```

Author: Done

1. **sds**:  used to enable [SDS (secret discovery service) for Istio](/docs/tasks/security/auth-sds). This profile comes only with authentication enabled.

The components marked as **X** are installed within each profile:

| Component | default | demo | minimal | remote | sds |
| --- | :---: | :---: | :---: | :---: | :---: |
|Profile filename | `values.yaml` | `values-istio-demo.yaml` | `values-istio-minimal.yaml` | `values-istio-remote.yaml` | `values-istio-sds-auth.yaml` |
| `istio-citadel` | X | X |   | X | X |
| Gateway |   |   |   |   |  |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-egressgateway` |   | X |   |   | |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-ingressgateway` | X | X |   |  | X |
| `istio-galley` | X | X |   |  | X |
| `istio-pilot` | X | X | X |   | X |
| Mixer |   |   |   |   |  |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-policy` | X | X |   |   | X |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-telemetry` | X | X |   |   | X |
| `istio-sidecar-injector` | X | X |   | X | X |
| `istio-nodeagent` |   |   |   |   | X |

> | Add on |   |   |   |   |   |

Reviewer: ```suggestion
| Add ons |   |   |   |   |   |
```

Author: Done. Had to add to .spelling and fixed up up another instance on this page.

| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`prometheus` | X | X |   |   | X |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`grafana` |   | X |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`istio-tracing` |   | X |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`kiali` |   | X |   |   |   |
|   |   |   |   |   |   |
| The authentication version, adding `-auth` adds |   |   |   |   |   |
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Control Plane Security |   | X |   |   |   |

> | &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mTLS |   | X |   |   | X |

Reviewer: ```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Strict mTLS |   | X |   |   | X |
```

Author: Done
